dask_sql/physical/rel/logical/aggregate.py

```diff
@@ -109,6 +110,9 @@ class LogicalAggregatePlugin(BaseRelPlugin):
         "max": AggregationSpecification("max", AggregationOnPandas("max")),
         "min": AggregationSpecification("min", AggregationOnPandas("min")),
         "single_value": AggregationSpecification("first"),
+        "regr_count": AggregationSpecification(
```

Comment:
This is just a normal "count", isn't it?
In this case, I would prefer to have 
```suggestion
        "regr_count": AggregationSpecification("count", AggregationOnPandas("count"))
```
as it might be faster than a "handmade" implementation. Or did I miss something?

Dask might be able to use some nice cython-optimized implementations on built-in methods. That might also be the case in your definition, but I would rely on the Dask developers to probably have a better importation here :-)

Reply:
And, one additional thought: can you include a comment why you can use a count here (basically, because you do the pre-transformation earlier)?

Reply:
Yes @nils-braun, I have tried that first, like using `AggregationSpecification("count", AggregationOnPandas("count"))` 
but got following error (refer below screenshot)
![image](https://user-images.githubusercontent.com/25167333/122611599-3acfb380-d09f-11eb-85cf-66f0449e475f.png)

This Error was raised when calling **regr_sxx,regr_syy** functions, As you already know our aggregation_dict looks like this 
```
# we are preparing this aggregation dict here in our code 
for aggregation in aggregations:
    input_col, output_col, aggregation_f = aggregation
    aggregations_dict[input_col][output_col] = aggregation_f
```

And Dask expects (aggregation_f, input_column) combination to be unique in the final dask graph plan, it makes sense because dask doesn't need to repeat the operation twice, And  our aggregation function produces (count,sepal_length ) twice this is why we are getting this error (for more reference into dask code base refer [here](https://github.com/dask/dask/blob/8aea537d925b794a94f828d35211a5da05ad9dce/dask/dataframe/groupby.py#L689)

To understand why  (count,sepal_length) was produced twice please refer to the below plan produced by the calcite 
![image](https://user-images.githubusercontent.com/25167333/122613544-a5362300-d0a2-11eb-9ad4-95cf3c9aad1b.png)

And since both normal count and regr_count using the same name **count** on `AggregationSpecification `, dask not allowing this duplicate combination I guess, and raising the error.  that's why I differentiated these two functions with a different name (count and regr_count). Do you know any other way to avoid this duplicate combination, may be calcite plan itself can avoid duplicate combinations by adding any optimization rule?  pleased to know your thoughts on this.

Reply:
Ah thanks, very detailed description.
However, I still think that using count would be a good idea and I actually think that this is a bug, because regr_count should not be applied on the bare column, but on your temporary column (created after applying not is nan). In your code, you are only doing the pre-processing if we have two inputs, but in the code example you are showing there is actually only one input to "regr_count". However, to be consistent, we should also create a temporary column with the not is nan pre-processing in this case. I guess all you need to do is change the if-elif logic.

Reply:
Thanks a lot for pointing out the bug and clearly explaining why it was Occurring, I think I have misunderstood the underlying problem now I am clear and able to implement this functionality with inbuilt dask functions instead of using custom functions, currently, I am using the following specification 

```
# is null was checked earlier, now only need to compute the sum of the non-null values
"regr_count": AggregationSpecification("sum", AggregationOnPandas("sum"))
```
if we are using built-in methods, I guess I need to change it into **sum** instead of **count** because, a sum only sums the boolean values (only True values) of the temporary column, and count counts both True and False

For regr_count function above implementation works and matches with the Postgres result.

Reply:
But After testing with nan introduced data frame, `regr_sxx` and `regr_syy` results are not exactly matching with Postgre's results and the testing formula (regr_count (y,x) * var_pop (y)), but matching with the relative tolerance to set to 1 in `assert_frame_equal`, is this okay to set this parameter, Dono Why I am getting that difference. Any idea here?